### PR TITLE
[merge] Feature: add a single student to the course API

### LINKED ISSUE #61: [feat] Create add student endpoint
In this Issue you will create a backend endpoint /api/students/post for creating a single student in a certain course.

This is required by PR #46 

Front end is not required for now but swagger is in urgent need

# Acceptance Criteria

- [x] The end point /api/students/post exists and does its work
- [x] It is shown on swagger page
- [x] It has test coverage

# Discussion

This will eventually be used on the Course Show Page #55

### PR DESCRIPTION
**Pull request by yuxiaolejs**
In this PR, I created endpoint for creating a single student in the course. 

## Pre-req
Nothing is needed before this branch gets merged. However, this branch can help the test of #46

# Feature
The endpoint is at `/api/students/post`, and shows as follows on swagger:

image

# User story
As an instructor, or teaching team member, I can add students to the repo without batch import. This is useful for students coming from waitlist. 

As a developer, I can test my course controller by adding individual dummy students (when used with the users API).

# Deployment
Our dokku server is running out of memory... Waiting for server

# Limitation
This feature does not have corresponding frontend pages created yet.

This is reqired by https://github.com/ucsb-cs156-s24/proj-organic-s24-5pm-3/pull/46

Closes #61

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/organic/controllers/StudentsController.java [hunk 1]
```diff
@@ -42,92 +47,116 @@
...
+        Course course = courseRepository.findById(courseId)
+                .orElseThrow(() -> new EntityNotFoundException(Course.class, courseId.toString()));
+
+        Iterable<Student> students = studentRepository.findByCourseId(course.getId());
+        return students;
+    }
+
+    @Operation(summary = "Create a new student for a course")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN','ROLE_INSTRUCTOR', 'ROLE_USER')")
+    // @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/controllers/StudentsController.java]: Commented out code...

**Review by pconrad — CHANGES_REQUESTED** [review]
One commented out line, otherwise LGTM

**Review by sherpatran — DISMISSED** [review]
LGTM besides the commented line

File: src/main/java/edu/ucsb/cs156/organic/controllers/StudentsController.java (hunk 1 above)
Comment (yuxiaolejs) [reply src/main/java/edu/ucsb/cs156/organic/controllers/StudentsController.java]: Removed!

**Review by tmcanepa — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**